src/components/CommentForm.jsx

```diff
@@ -0,0 +1,71 @@
+import { useState } from "react";
+import { useParams } from "react-router-dom";
+import { postComment } from "../api";
+
+function CommentForm({ setComments, comments }) {
+  const [commentBody, setCommentBody] = useState("");
+  const { article_id } = useParams();
+  const [isPosting, setIsPosting] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [isSuccessful, setIsSuccessful] = useState(false);
+  const [isValidComment, setIsValidCOmment] = useState(true)
+
+  const handleBodyChange = (e) => {
+    setCommentBody(e.target.value);
+  };
+
+ 
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    setIsValidCOmment(true)
+    setIsSuccessful(false);
+    if (commentBody.length === 0) {
+        setIsValidCOmment(false)
+        return
+      }
+    setIsError(false);
+    setIsPosting(true);
+    setCommentBody(e.target[0].value);
+    const bodyForPosting = {
+      article_id: article_id,
+      username: "tickle122",
+      body: commentBody,
+    };
+    postComment(article_id, bodyForPosting)
+      .then((response) => {
+        const newComment = response.data.comment;
+        setIsPosting(false);
+        setComments([newComment, ...comments]);
+        setCommentBody("");
+        setIsSuccessful(true);
+      })
+      .catch((err) => {
+        setIsError(true);
+      });
+  };
+
+  return (
+    <div className="comment-form">
+      <form onSubmit={handleSubmit}>
+        <label htmlFor="comment-text" className="bold">Enter a comment below:</label>
```

Comment:
Depends on your design but given API functionality as an extension you might want to consider making the ability to leave a comment conditional based on whether or not the user is logged in later on (assuming you don't want to use a hardcoded user - we'll look at a way to do this in the lecture tomorrow so this isn't really feedback, more just a suggestion to future Adam to take another look at this later on 😄)